### PR TITLE
React router2

### PR DESCRIPTION
**Pull request by CaptainAngus**
Add show page with reviews, comments, etc.

### CODE REVIEW

File: app/javascript/react/src/containers/CommentContainer.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  // componentDidMount() {
+  //   fetch('/api/v1/reviews/:review_id/comments/')
+  //   .then(response => response.json())
+  //   .then(body => {
+  //     this.setState({ comments: body })
+  //   })
+  // }
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/containers/CommentContainer.js]: Lines 13 - 19 and 33 Has commented out code. We shouldn't be merging this with master.

File: app/javascript/react/src/containers/CommentContainer.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react"
+import CommentTile from '../components/CommentTile'
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/containers/CommentContainer.js]: Semi-colons (lines 1 + 2)

File: app/javascript/react/src/containers/IndexPage.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import IndexSearch from './IndexSearch'
+import NavBar from './NavBar'
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/containers/IndexPage.js]: Semi-colons (lines 1 - 3)

File: app/javascript/react/src/containers/RepoShowContainer.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react"
+import RepoShowComponent from "../components/RepoShowComponent"
+import ReviewContainer from "../containers/ReviewContainer"
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/containers/RepoShowContainer.js]: Semi-colons (lines 1 - 3)

File: app/javascript/react/src/containers/ReviewContainer.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    .then(response => reponse.json)
+    .then(responseData => {
+      this.setState({ reviews: [...this.state.reviews, responseData] })
+    })
+  }
+
+  render() {
+
+    // let addNewReview=(formPayload) => this.addNewReview(formPayload)
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/containers/ReviewContainer.js]: Commented out code on line 36 and 46

File: app/javascript/react/test/components/ReviewTileTest.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  ]
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <ReviewTile review={review}/>
+    )
+  })
+
+  it('should render')
+})
```
Comment (LiannaNewman) [inline app/javascript/react/test/components/ReviewTileTest.js]: Is this test incomplete? It should be left out of the PR.

File: config/routes.rb [hunk 7]
```diff
@@ -30,4 +30,6 @@
       #get ":user_slug/:repo_slug/reviews/:id/comments", to: ""
     end
   end
+
+  get "*path", to: "search#index"
```
Comment (LiannaNewman) [inline config/routes.rb]: Looks like there's a typo on line 34 `*path`?

File: spec/helpers/show_helper_spec.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# describe ShowHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe ShowHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (LiannaNewman) [inline spec/helpers/show_helper_spec.rb]: This looks like an extra unused file please remove from PR request.

File: app/helpers/show_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module ShowHelper
+end
```
Comment (LiannaNewman) [inline app/helpers/show_helper.rb]: This looks like an extra unused file please remove from PR request.

File: app/javascript/react/src/components/CommentTile.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react"
+import TextInputField from '../components/TextInputField'
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/components/CommentTile.js]: Semi-colons lines 1-2

File: app/javascript/react/src/components/ReviewTile.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react"
+import TextInputField from './TextInputField'
+import CommentContainer from '../containers/CommentContainer'
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/components/ReviewTile.js]: Semi-colon lines 1 -3

File: app/javascript/react/src/components/TagTile.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { Link } from 'react-router'
+
```
Comment (LiannaNewman) [inline app/javascript/react/src/components/TagTile.js]: Semi-colo line 2

File: config/routes.rb (hunk 7 above)
Comment (CaptainAngus) [reply config/routes.rb]: This is working as intended.

**Comment by CaptainAngus** [comment]
Made all but one of the changes- commented on that one. Thanks!

**Comment by CaptainAngus** [comment]
Ok, NOW it's ready!

File: app/javascript/react/src/containers/NavBar.js [hunk 13]
```diff
@@ -23,10 +23,6 @@ class NavBar extends Component {
 
     fetch(`/users/${sign_in_or_out}`, {
       method: `${method}`
-      // headers: {
-      //   'Content-Type': 'application/json'
-      // },
-      // body: JSON.stringify(content)
     })
```
Comment (chrissyfancy) [inline app/javascript/react/src/containers/NavBar.js]: What's going on here?

File: app/javascript/react/test/containers/CommentContainer.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+  });
+
+  // it('should should have the specified initial state', () => {
+  //   expect(wrapper.state()).toEqual({ comments: [] });
+  // });
+
+  // it('should render an CommentContainer', () => {
+  //   expect(wrapper.find(CommentContainer)).toBePresent();
+  // });
+
```
Comment (chrissyfancy) [inline app/javascript/react/test/containers/CommentContainer.js]: Add to these or remove them!

**Pull request merged**